arrow-fx-coroutines/src/test/kotlin/arrow/fx/coroutines/ParTraverseTest.kt

```diff
@@ -160,4 +172,24 @@ class ParTraverseTest : ArrowFxSpec(spec = {
     val l = (0 until count).parTraverseN(20) { it }
     l shouldBe (0 until count).toList()
   }
+
+  "parSequenceN throws validation error as semaphore limit is greater than Int max value" {
+    val ref = Atomic(0)
+    shouldThrowExactly<ArithmeticException> {
+      (0 until 100)
+        .map { i -> suspend { ref.update { i + 1 } } }
+        .parSequenceN(Long.MAX_VALUE)
+
+      ref.get() shouldBe 100
+    }
+  }
+
+  "parSequenceN can traverse effect full computations" {
+    val ref = Atomic(0)
+    (0 until 100)
+      .map { i -> suspend { ref.update { i + 1 } } }
```

Comment:
```suggestion
      .map { suspend { ref.update { it + 1 } } }
```

Reply:
Just for my understanding, why prefer `it`, doesn't it make it harder to infer where it is coming from ?

Reply:
You were using the wrong lambda parameter. The following is correct with named params.

`map { _ -> suspend { ref.update { i -> i + 1 } } }`

Reply:
I am stupid! 😀 should've seen that. 

The problem is I am running everything with gradle daemon, as kotlin compiler in intellij just goes OOM, even when i've given 3 gigs to it. 
And with gradle, with every change i've to build first and then execute. 
If I forget to do that, it just runs on previous changes, and eludes me into thinking it has executed latest changes

Reply:
> The problem is I am running everything with gradle daemon, as kotlin compiler in intellij just goes OOM, even when i've given 3 gigs to it.

There is an option to build with IntelliJ, or delegate building to Gradle. I've set it to Gradle that fixes this issue.

> If I forget to do that, it just runs on previous changes, and eludes me into thinking it has executed latest changes

Oh yes.. super annoying. I started only running tests from Gradle CLI for that reason 😅